crmsh/bootstrap.py

```diff
@@ -228,13 +231,49 @@ def wait_for_cluster():
     status_long("Waiting for cluster")
     while True:
         _rc, out, _err = utils.get_stdout_stderr("crm_mon -1")
-        if "online" in out.lower():
+        if is_online(out):
             break
         status_progress()
-        sleep(5)
+        sleep(2)
     status_done()
 
 
+def get_peer_hostname():
+    """
+    In join process, cluster_node might be an IP or hostname
+    Return hostname anyway
+    """
+    peer_node = None
+    if _context.cluster_node:
+        if utils.valid_ip_addr(_context.cluster_node):
+            peer_node = utils.get_stdout("ssh {} hostname".format(_context.cluster_node))[1]
+        else:
+            peer_node = _context.cluster_node
+    return peer_node
+
+
+def is_online(crm_mon_txt):
+    """
+    Check whether local node is online
+    Besides that, in join process, check whether init node is online
+    """
+    if not re.search("Online: .* {} ".format(utils.this_node()), crm_mon_txt):
+        return False
+
+    peer_node = get_peer_hostname()
```

Comment:
I would directly return True here if peer_node is False. It would make the code easier to read.
And we could add comments.
If peer_node is False, it means that this is a `init` process, right? I would add this comments there

Reply:
OK, changed